lib/main_publiccloud.pm

```diff
@@ -0,0 +1,118 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: module loader of container tests
+# Maintainer: qa-c@suse.de
+
+package main_publiccloud;
+use Mojo::Base 'Exporter';
+use utils;
+use version_utils;
+use main_common qw(loadtest load_extra_tests_prepare);
+use testapi qw(check_var get_var);
+
+our @EXPORT = qw(
+  load_publiccloud_tests
+);
+
+sub load_podman_tests() {
+    loadtest 'containers/podman';
+    loadtest 'containers/podman_3rd_party_images';
+}
+
+sub load_docker_tests() {
+    loadtest 'containers/docker';
+    loadtest 'containers/docker_runc';
+    loadtest 'containers/docker_image';
+    loadtest 'containers/docker_3rd_party_images';
+    loadtest 'containers/registry';
+    loadtest 'containers/zypper_docker';
+}
+
+sub load_qam_publiccloud_tests {
+    my $args = OpenQA::Test::RunArgs->new();
+
+    loadtest "publiccloud/download_repos";
+    loadtest "publiccloud/prepare_instance", run_args => $args;
+    loadtest "publiccloud/register_system", run_args => $args;
+    loadtest "publiccloud/transfer_repos", run_args => $args;
+    loadtest "publiccloud/patch_and_reboot", run_args => $args;
+    if (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
+        loadtest("publiccloud/img_proof", run_args => $args);
+    } elsif (get_var('PUBLIC_CLOUD_LTP')) {
+        loadtest('publiccloud/run_ltp', run_args => $args);
+    } elsif (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')) {
+        loadtest('publiccloud/boottime', run_args => $args);
+    } elsif (get_var('PUBLIC_CLOUD_FIO')) {
+        loadtest('publiccloud/storage_perf', run_args => $args);
+    } else {
+        loadtest "publiccloud/ssh_interactive_start", run_args => $args;
+        loadtest "publiccloud/instance_overview" unless get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+        if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {
+            load_extra_tests_prepare();
+            load_publiccloud_consoletests();
+        }
+        if (get_var('PUBLIC_CLOUD_CONTAINERS')) {
+            load_podman_tests() if is_sle('>=15-sp1');
```

Comment:
you skip `load_extra_tests_prepare`  in comparison with version which was deleted in `main_common` . was it intentional and why if it was intentional ?

Reply:
Probably not, see the [thread](https://suse.slack.com/archives/C02CGKBCGT1/p1635156292116700) where @grisu48 pointed that this module also checks that the maintenance updates are downloaded. But I'd maybe move such check to `transfer_repos.pm`. Let's create a ticket for that?

Reply:
When i came to it i assumed that the containers yaml scheduling has obsoleted the function in *main_common*. *load_qam_publiccloud_tests* was not following the updates in the yaml IMO. i dont know but this is what i pressume. that said the *PUBLIC_CLOUD_CONTAINERS* var should schedule only the containers. And currently, `publiccloud_containers` jobs do not contain anything from `load_extra_tests_prepare`.

Reply:
yeah sure , I would use same assumptions . Just wanted to make sure that this is intentional , because you moved code to different place it is not obvious from the diff that you also dropped some code on the way